ConsoleApp1/Program.cs

```diff
@@ -232,8 +146,199 @@ static void SearchRecords(string path)
+			Console.Write("Enter record index (0 to cancel): ");
 
-			//DisplayPhoneBook(searchedRecords);
+			int index;
+
+			if (int.TryParse(Console.ReadLine() ?? "1", out index))
+			{
+				if (index > 0)
+				{
+					(string firstName, string lastName, string phone)[] phoneBookRecords = ReadPhoneBookRecords(path);
+
+                    if (index - 1 < phoneBookRecords.Length)
+					{
+                        (string firstName, string lastName, string phone)[] res = 
+							new (string firstName, string lastName, string phone)[phoneBookRecords.Length - 1];
+
+                        for (int i = 0; i < phoneBookRecords.Length; i++)
+						{
+							if (i < index - 1)
+							{
+								res[i] = phoneBookRecords[i];
+
+							} else if (i > index - 1)
+							{
+								res[i - 1] = phoneBookRecords[i];
+							}
+						}
+
+						string[] strRes = new string[res.Length  + 1];
+						strRes[0] = "FirstName,LastName,PhoneNumber";
+
+                        for (int i = 0; i < res.Length; i++)
+						{
+							strRes[i + 1] = $"{res[i].firstName},{res[i].lastName},{res[i].phone}";
+						}
+                        File.WriteAllLines(path, strRes);
+					}
+				}
+
+            } else
+			{
+				Console.WriteLine("invalid input");
+			}
+        }
+
+		static void SortRecords(string path)
+		{
+
+            Func<(string firstName, string lastName, string phone), int, string> SortBy = 
+				(a, b) => b == 1 ? a.firstName : b == 2 ? 
+					a.lastName : a.phone;
+			Func<string, string, int, bool> sorter = (a, b, o) => o == 1 ?
+				string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 ? false : true : 
+				string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 ? true : false;
+
+            (string firstName, string lastName, string phone)[] rec = ReadPhoneBookRecords(path);
+
+            Console.WriteLine("\t1. first name");
+            Console.WriteLine("\t2. last name");
+            Console.WriteLine("\t3. phone number");
+            Console.Write("Sort by: ");
+            var pressedKey = Console.ReadKey();
+			int sortBy = 1;
+			switch (pressedKey.Key)
+			{
+				case ConsoleKey.D1:
+					sortBy = 1;
+					break;
+				case ConsoleKey.D2:
+					sortBy = 2;
+					break;
+				case ConsoleKey.D3:
+					sortBy = 3;
+					break;
+				default:
+					sortBy = -1;
+					break;
+			}
+
+			Console.WriteLine();
+            Console.WriteLine("\t1. ASC");
+			Console.WriteLine("\t2. DSC");
+            Console.Write("Sort order: ");
+
+            pressedKey = Console.ReadKey();
+            int order = 1;
+            switch (pressedKey.Key)
+            {
+                case ConsoleKey.D1:
+                    order = 1;
+                    break;
+                case ConsoleKey.D2:
+                    order = 2;
+                    break;
+                default:
+                    order = -1;
+                    break;
+            }
+
+			if(sortBy != -1 && order != -1)
+			{
+				for (int i = 1; i < rec.Length; i++)
```

Comment:
Sort logic is duplicated, create method Sort and pass all required parameters, and reuse below